grant-project/zcash-grant-system
# Contribution expired notification

Part of #198.

### What This Does

Creates a new task that gets made on contribution creation. 24h after the contribution is made, if it's still pending, emails the user to indicate the contribution has expired. Doesn't change anything in the database, simply sends the notification. This of course does not work on anonymous contributions.

Maybe we should add a new `contribution.status` for expired contributions instead of it being kind of implicit, let me know what y'all think.

### Steps to Test

* Create a contribution, confirm a new row shows up in the `task` table
* Change the execute_after date to be in the past and hit `http://localhost:5000/api/v1/task`, confirm you get an expiry email
* Create an anonymous contribution, confirm no new `task` row is created

## Screenshots

<img width="627" alt="screen shot 2019-03-06 at 3 31 27 pm" src="https://user-images.githubusercontent.com/649992/53911862-a8b65500-4025-11e9-868a-62ff3bb31d16.png">
